nsot/models.py

```diff
@@ -1396,8 +1396,43 @@ def set_addresses(self, addresses, overwrite=False, partial=False):
 
         self.clean_addresses()
 
+    def get_ancestors(self):
+        """Recursively get all parents of an interface"""
+        p = self.parent
+        ancestors = []
+        while p is not None:
+            ancestors.append(p)
+            p = p.parent
+        return ancestors
+
+    def get_children(self):
+        """Return the immediate children of an interface"""
+        return Interface.objects.filter(parent=self)
+
+    def get_descendants(self):
+        """Recursively return all the children of an interface"""
+        s = list(self.get_children())
+        children = []
+        while len(s) > 0:
+            top = s.pop()
+            children.append(top)
+            for c in top.get_children():
+                s.append(c)
+        return children
+
+    def get_root(self):
+        """Return the parent of all ancestors of an interface"""
+        root = self.parent
+        while root is not None and root.parent is not None:
+            root = root.parent
+        return [] if root is None else [root]
```

Comment:
Similar to @jathanism's [comment on `parent`](https://github.com/dropbox/nsot/pull/271/files#r116053169), this should have the same behavior and return a single object rather than a list.